app/controllers/application_controller.rb

```diff
@@ -45,6 +44,12 @@ def create_user_event(event_type, user = current_user)
 
   private
 
+  def redirect_on_timeout
+    return unless request.query_parameters[:timeout]
+    flash[:timeout] = t('notices.session_cleared')
+    redirect_to request.path
```

Comment:
Do you think it's worth the effort to try to keep the request of the query parameters in the URL? Not sure if any of our pages keep useful stuff in the URL

ex

```ruby
redirect_to request.path + '?' + query_parameters.except(:timeout).to_query
```

except maybe something even fancier to drop the `?` entirely if the query string is blank?